adding carts route handlers

TheLastDance opened the pull request:
- added increment, get-cart, get-carts and create-cart route handlers
- implemented logic of first addition to cart and increment quantity
- created api endpoint get-cart which will do all algorithms at backend side and then return fresh products data with all calculations
- added cart page for test purposes
- added CartTable type

TheLastDance committed:
adding carts route handlers

- added increment, get-cart, get-carts and create-cart route handlers
- implemented logic of first addition to cart and increment quantity
- created api endpoint get-cart which will do all algorithms at backend side and then return fresh products data with all calculations
- added cart page for test purposes
- added CartTable type

pllague commented:
Please move routes from carts directory to api

pllague commented on src/app/[locale]/(dashboard)/cart/page.tsx:
Everything works well, just delete "src/app/[locale]/(dashboard)/cart/page.tsx" file, i have already created "src/app/[locale]/(dashboard)/checkout/page.tsx" page as cart page

pllague commented on src/app/api/carts/get-cart/route.ts:
Move "fetchData" function to the "helpers.ts" file and import it here.

TheLastDance replied:
Maybe it will be a better idea to move this fetch data function to api.ts? Because we use it for fetch purposes?

pllague replied:
Yee correct, do it 👍

TheLastDance committed:
SES-19-David refactoring code

- deleted cart page
- moved fetch method to api.ts
- moved route handlers to /api folder
